apps/backend/src/action/logic/pass.ts

```diff
@@ -1,25 +1,55 @@
 import { ResultAsync, errAsync } from "neverthrow";
 import { EventSource } from "~/models/event";
 import { PassCommand } from "~/action/command";
-import { TurnHasBeenPassedEvent, Event } from "~/action/event";
+import { TurnHasBeenPassedEvent } from "~/action/event";
+import { always, error } from "~/utils";
+
+export type RepositoryWriteError = ReturnType<typeof RepositoryWriteError>;
+export type PassError = RepositoryWriteError;
 
 export interface Pass {
-  (source: EventSource<Event>, command: PassCommand): ResultAsync<
-    TurnHasBeenPassedEvent,
-    Error
-  >;
+  (
+    source: EventSource<TurnHasBeenPassedEvent>,
+    command: PassCommand
+  ): ResultAsync<TurnHasBeenPassedEvent[], Error /* PassError */>;
 }
 
+const RepositoryWriteError = error("RepositoryWriteError");
+
+const appendEventToEventSource = (
+  source: EventSource<TurnHasBeenPassedEvent>,
+  command: PassCommand
+) =>
+  ResultAsync.fromPromise(
+    source.append(TurnHasBeenPassedEvent(command.data.card)),
+    always(RepositoryWriteError("failed to write event to repository"))
+  );
+
 /**
  * *description*
  * passes the turn to the next player.
  *
+ * trigger situation:
+ * 1. no hands
+ * 2. discard card
+ *
  * *param* source - event source
  * *param* command - pass command
  */
-export const pass: Pass = (source, command) => {
-  //@todo: append event to event source
-  return errAsync(new Error("not implemented"));
-};
+export const pass: Pass = (source, command) =>
+  // @todo should validate user hands
+  /**
+   *
+   * 1. check player have cards left
+   *  if player not cards left and discard card eq null
+   *    - can Pass
+   *
+   * 2. check player have the card
+   *  get player hands if hands can find the discard card
+   *    - can Pass
+   *  else not find
+   *    - Error
+   **/
```

Comment:
God Job!!!
Try to use English, and don't be afraid to make mistakes.

You can also consider using ChatGPT or Grammarly to check your sentence.
This definitely helps.